hudi-client/hudi-client-common/src/main/java/org/apache/hudi/keygen/NonpartitionedAvroKeyGenerator.java

```diff
@@ -45,6 +51,18 @@ public String getPartitionPath(GenericRecord record) {
     return EMPTY_PARTITION_FIELD_LIST;
   }
 
+  @Override
+  public String getRecordKey(GenericRecord record) {
+    // the format of record key varies due to the number of record key fields
+    // 1. if there is only one record key field, the format of record key is just "<value>"
+    // 2. if there are multiple record key fields, the format is "<field1>:<value1>,<field2>:<value2>,..."
+    // for backward compatibility, we need to use the right format according to the number of record key fields
```

Comment:
nit: Can you move this to beginning (line 56. you can remove "the format of record key varies due to the number of record key fields"). I also prefer line 63 to be in else block (in some cases, compiler optimization works better that way. This is a 'nit' comment, so Its fine if you want to leave it as is.)

Reply:
Updated the code according to your suggestion.